pysph/base/opencl.py

```diff
@@ -35,6 +35,47 @@ def set_queue(q):
     _queue = q
 
 
+class DeviceArray(cl.array.Array):
+    def __init__(self, queue, dtype, n=0):
+        self.queue = queue
+        self.length = n
+        if n == 0:
+            n = 16
+        self.alloc = n
+        self.dtype = dtype
+        self._array = cl.array.empty(queue, n, dtype)
+
+    @property
+    def data(self):
+        return self._array[:self.length]
```

Comment:
I would suggest that the view only be created when the length changes, otherwise you are still creating a new object every time someone does `dev_array.data`.